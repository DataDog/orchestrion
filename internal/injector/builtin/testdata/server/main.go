// Unless explicitly stated otherwise all files in this repository are licensed
// under the Apache License Version 2.0.
// This product includes software developed at Datadog (https://www.datadoghq.com/).
// Copyright 2023-present Datadog, Inc.

package main

import (
	"io"
	"log"
	"net/http"
//line <generated>:1
	"github.com/datadog/orchestrion/instrument"
)

//line samples/server/main.go:14
func main() {
	s := &http.Server{
		Addr: ":8080",
		Handler:
//line <generated>:1
		//dd:startwrap
		instrument.WrapHandler(
//line samples/server/main.go:17
			http.HandlerFunc(myHandler)),
		//dd:endwrap
	}

//line samples/server/main.go:20
	log.Fatal(s.ListenAndServe())
}

// myHandler comment on function
func myHandler(w http.ResponseWriter, r *http.Request) {
//line <generated>:1
	//dd:startinstrument
	{
		instrument.Report(r.Context(), instrument.EventStart, "name", "myHandler", "verb", r.Method)
		defer instrument.Report(r.Context(), instrument.EventEnd, "name", "myHandler", "verb", r.Method)
	}
	//dd:endinstrument
//line samples/server/main.go:25
	b, err := io.ReadAll(r.Body)
	// test comment in function
	if err != nil {
		w.WriteHeader(http.StatusBadRequest)
		w.Write([]byte(err.Error()))
		return
	}
	defer r.Body.Close()
	w.WriteHeader(http.StatusOK)
	w.Write(b)
}

func instrumentedHandler(w http.ResponseWriter, r *http.Request) {
//line <generated>:1
	//dd:startinstrument
	{
		instrument.Report(r.Context(), instrument.EventStart, "name", "instrumentedHandler", "verb", r.Method)
		defer instrument.Report(r.Context(), instrument.EventEnd, "name", "instrumentedHandler", "verb", r.Method)
	}
	//dd:endinstrument
//line samples/server/main.go:38
	b, err := io.ReadAll(r.Body)
	// test comment in function
	if err != nil {
		w.WriteHeader(http.StatusBadRequest)
		w.Write([]byte(err.Error()))
		return
	}
	defer r.Body.Close()
	w.WriteHeader(http.StatusOK)
	w.Write(b)
}

// comment that is just hanging out unattached
