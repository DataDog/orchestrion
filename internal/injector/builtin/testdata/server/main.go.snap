//line samples/server/main.go:1:1
// Unless explicitly stated otherwise all files in this repository are licensed
// under the Apache License Version 2.0.
// This product includes software developed at Datadog (https://www.datadoghq.com/).
// Copyright 2023-present Datadog, Inc.

package main

import (
	"io"
	"log"
	"net/http"
//line <generated>:1
	__orchestrion_tracer "gopkg.in/DataDog/dd-trace-go.v1/ddtrace/tracer"
	__orchestrion_profiler "gopkg.in/DataDog/dd-trace-go.v1/profiler"
	__orchestrion_os "os"
)

//line samples/server/main.go:14
func main() {
//line <generated>:1
	{
		defer __orchestrion_tracer.Stop()
		defer __orchestrion_profiler.Stop()
	}
//line samples/server/main.go:15
	s := &http.Server{
		Addr:    ":8080",
		Handler: http.HandlerFunc(myHandler),
	}

	log.Fatal(s.ListenAndServe())
}

// myHandler comment on function
func myHandler(w http.ResponseWriter, r *http.Request) {
	b, err := io.ReadAll(r.Body)
	// test comment in function
	if err != nil {
		w.WriteHeader(http.StatusBadRequest)
		w.Write([]byte(err.Error()))
		return
	}
	defer r.Body.Close()
	w.WriteHeader(http.StatusOK)
	w.Write(b)
}

func instrumentedHandler(w http.ResponseWriter, r *http.Request) {
	b, err := io.ReadAll(r.Body)
	// test comment in function
	if err != nil {
		w.WriteHeader(http.StatusBadRequest)
		w.Write([]byte(err.Error()))
		return
	}
	defer r.Body.Close()
	w.WriteHeader(http.StatusOK)
	w.Write(b)
}

// comment that is just hanging out unattached
//
//line <generated>:1
func init() {
	__orchestrion_tracer.Start(__orchestrion_tracer.WithOrchestrion(map[string]string{"version": "<version.Tag>"}))
}
func init() {
	switch __orchestrion_os.Getenv("DD_PROFILING_ENABLED") {
	case "1", "true", "auto":
		// The "auto" value can be set if profiling is enabled via the
		// Datadog Admission Controller. We always turn on the profiler in
		// the "auto" case since we only send profiles after at least a
		// minute, and we assume anything running that long is worth
		// profiling.
		err := __orchestrion_profiler.Start(
			__orchestrion_profiler.WithProfileTypes(__orchestrion_profiler.CPUProfile, __orchestrion_profiler.HeapProfile, __orchestrion_profiler.GoroutineProfile, __orchestrion_profiler.MutexProfile),
			__orchestrion_profiler.WithTags("orchestrion:true"),
		)
		if err != nil {
			// TODO: is there a better reporting mechanism?
			// The tracer and profiler already use the stdlib logger, so
			// we're not adding anything new. But users might be using a
			// different logger.
			log.Printf("failed to start profiling: %s", err)
		}
	}
}
